### PR TITLE
test(e2e): readme

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
Adds a bit of doc about e2e tests, so first-time contributors are less confused.

### Checklist prior to review

- [ ] Link to docs PR or issue --
- [ ] Link to UI issue or PR --
- [ ] Is the [issue worked on linked][1]? --
- [ ] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [ ] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch)?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e_env/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Every test suite should run in their own mesh and namespace to not interfere with other tests.
+
+
+## Recommendations
+
+* (Kubernetes): Use `TriggerDeleteNamespace` instead of `DeleteNamespace`.
+  It takes a lot of time to wait for the namespace to be removed.
+  We should carry on with other tests while namespace is removed in the background.
+* (Kubernetes): Be conservative about resource limits on Kubernetes.
+  We are deploying a lot of stuff in a parallel, default `DemoClient` and `TestServer` are tuned to have minimal requests/limit.
```
Comment (lahabana) [inline test/e2e_env/README.md:26]:
```suggestion
  We are deploying a lot of stuff in parallel, default `DemoClient` and `TestServer` are tuned to have minimal requests/limit.
```

File: test/e2e_env/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  If you are bringing new deployment to the test, make sure you set a proper resource limits.
+* `KUMA_STORE_UNSAFE_DELETE` is set to true by default, so we can remove the mesh without waiting for all DPPs to be down.
+* (Multizone) Remember about calling `WaitForMesh`, otherwise deploying an application in Zone can fail.
+
+## Limitations
+
+* (Universal) Getting logs in the test is currently not supported.
+* (Universal) UniversalCluster#apps is not shared across parallelized processes.
```
Comment (lahabana) [inline test/e2e_env/README.md:34]: Should you add links to issues?

**Review by lahabana — APPROVED** [review]
Thx!

File: test/e2e_env/README.md (hunk 2 above)
Comment (jakubdyszkiewicz) [reply test/e2e_env/README.md:34]: I don't think we ever want to fix the not synced apps. Put a link for logging.

**Pull request merged**